pkg/agent/handler/environment.go

```diff
@@ -0,0 +1,48 @@
+package handler
+
+import (
+	"context"
+
+	agentcache "github.com/Axway/agent-sdk/pkg/agent/cache"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	environment "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	management "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	"github.com/Axway/agent-sdk/pkg/config"
+	"github.com/Axway/agent-sdk/pkg/watchmanager/proto"
+)
+
+type environmentHandler struct {
+	agentCacheManager agentcache.Manager
+	credentialConfig  config.CredentialConfig
+}
+
+// NewEnvironmentHandler creates a Handler for Environments.
+func NewEnvironmentHandler(agentCacheManager agentcache.Manager, credentialConfig config.CredentialConfig) Handler {
+	return &environmentHandler{
+		agentCacheManager: agentCacheManager,
+		credentialConfig:  credentialConfig,
+	}
+}
+
+func (c *environmentHandler) Handle(ctx context.Context, meta *proto.EventMeta, resource *v1.ResourceInstance) error {
+	action := GetActionFromContext(ctx)
+	if resource.Kind != environment.EnvironmentGVK().Kind {
+		return nil
+	}
+
+	if action == proto.Event_SUBRESOURCEUPDATED && meta.Subresource == management.EnvironmentPoliciesSubResourceName {
```

Comment:
Can you change this to return nil if action is not subresource or subresource not policies? That way the meat of the function is not within the fi statement?